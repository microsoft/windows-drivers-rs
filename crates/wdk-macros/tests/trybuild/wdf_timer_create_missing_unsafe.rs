// Copyright (c) Microsoft Corporation
// License: MIT OR Apache-2.0
#![no_main]
#![deny(warnings)]

use wdk_sys::*;

fn foo(timer_config: &mut WDF_TIMER_CONFIG, attributes: &mut WDF_OBJECT_ATTRIBUTES,) {
    let mut timer = core::ptr::null_mut();
    let _nt_status = macros::call_unsafe_wdf_function_binding!(
        WdfTimerCreate,
        timer_config,
        attributes,
        &mut timer,
    );
}
