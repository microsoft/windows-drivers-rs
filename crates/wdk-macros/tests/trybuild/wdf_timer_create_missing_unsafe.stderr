error[E0133]: call to unsafe function is unsafe and requires unsafe function or block
 --> tests/trybuild/wdf_timer_create_missing_unsafe.rs
  |
  |       let _nt_status = macros::call_unsafe_wdf_function_binding!(
  |  ______________________^
  | |         WdfTimerCreate,
  | |         timer_config,
  | |         attributes,
  | |         &mut timer,
  | |     );
  | |_____^ call to unsafe function
  |
  = note: consult the function's documentation for information on how to avoid undefined behavior
  = note: this error originates in the macro `macros::call_unsafe_wdf_function_binding` (in Nightly builds, run with -Z macro-backtrace for more info)
